### PR TITLE
[VM] Handle the Discard type of status code from native function execution

### PR DESCRIPTION
**Pull request by steelgeek091**

### CODE REVIEW

File: frameworks/moveos-stdlib/src/natives/helpers.rs [hunk 1]
```diff
@@ -58,7 +62,22 @@ where
...
+            if let Err(err) = &result {
+                let status = err.major_status();
+                let vm_status = VMStatus::error(status, Some("".to_string()));
+                match vm_status.keep_or_discard() {
+                    Ok(_) => result,
+                    Err(_) => {
+                        log::error!("Native function execution failed {}", err.to_string());
+                        Ok(NativeResult::err(
+                            InternalGas::new(3000),
+                            E_NATIVE_FUNCTION_EXECUTING_PANIC,
```
Comment (steelgeek091) [inline frameworks/moveos-stdlib/src/natives/helpers.rs:74]: Return `NativeResult::err` in there appropriate or not?

File: frameworks/moveos-stdlib/src/natives/helpers.rs [hunk 2]
```diff
@@ -58,7 +62,22 @@ where
                     },
                 }
             }
-            result
+            if let Err(err) = &result {
+                let status = err.major_status();
+                let vm_status = VMStatus::error(status, Some("".to_string()));
+                match vm_status.keep_or_discard() {
```
Comment (jolestar) [inline frameworks/moveos-stdlib/src/natives/helpers.rs:68]:
```suggestion
                let vm_status = VMStatus::error(status, None);
```

File: frameworks/moveos-stdlib/src/natives/helpers.rs [hunk 3]
```diff
@@ -58,7 +62,22 @@ where
...
+                    Ok(_) => result,
+                    Err(_) => {
+                        log::error!("Native function execution failed {}", err.to_string());
+                        Ok(NativeResult::err(
```
Comment (jolestar) [inline frameworks/moveos-stdlib/src/natives/helpers.rs:72]: We should convert the discard VMError to a keept VMError, not NativeResult::err

Comment (steelgeek091) [reply frameworks/moveos-stdlib/src/natives/helpers.rs]: The return type of this function is `PartialVMResult`, but the struct `PartialVMError` does not have the `Location` information in it:

```rust
struct PartialVMError_ {
    major_status: StatusCode,
    sub_status: Option<u64>,
    message: Option<String>,
    exec_state: Option<ExecutionState>,
    indices: Vec<(IndexKind, TableIndex)>,
    offsets: Vec<(FunctionDefinitionIndex, CodeOffset)>,
}
```

**Review by jolestar — APPROVED** [review]

**Pull request merged**